### PR TITLE
Ensure plugin can be installed and tests can be run

### PR DESCRIPTION
**Pull request by justinmayer**
Refs https://github.com/getpelican/cookiecutter-pelican-plugin/issues/3

### CODE REVIEW

**Comment by justinmayer** [comment]
@davidag: Can you tell me whether this resolves the problem you reported in the above-linked issue? If so, I'll make the same changes to the plugin template.

**Review by avaris — APPROVED** [review]

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -14,13 +14,19 @@ To set up Feed Filter for local development, first install [Poetry][]:
...
+
+Clone the Pelican repository:
 
-The last two steps will install the needed dependencies and set up the project:
+    git clone https://github.com/getpelican/pelican.git ~/Projects/
+
+Install the needed dependencies and set up the project:
 
     poetry install
+    pip install U -e ../../pelican
```
Comment (davidag) [inline CONTRIBUTING.md]: I think you meant `-U`.

**Review by davidag — CHANGES_REQUESTED** [review]
It works now :tada:

File: CONTRIBUTING.md (hunk 1 above)
Comment (justinmayer) [reply CONTRIBUTING.md]: Indeed. Nice catch! I removed that flag in favor of another approach.

**Comment by justinmayer** [comment]
@davidag: I made a bunch of other improvements to the documentation. Let me know what you think. 😊

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -14,12 +14,23 @@ To set up Feed Filter for local development, first install [Poetry][]:
 
 Go to the [Feed Filter repository][] and tap the **Fork** button at top-right. Then clone the source for your fork and add the upstream project as a Git remote:
 
-    git clone https://github.com/YOUR_USERNAME/feed-filter.git
-    cd feed-filter
-    git remote add upstream feed-filter.git
+    mkdir -p ~/projects/pelican-plugins ~/virtualenvs
```
Comment (davidag) [inline CONTRIBUTING.md]: I would create `~/virtualenvs` right before using it, ie. in line 27. It seems clearer to me.

Comment (davidag) [reply CONTRIBUTING.md]: Same thing in the detailed set-up.

**Comment by davidag** [comment]
> @davidag: I made a bunch of other improvements to the documentation. Let me know what you think. blush

@justinmayer Great! Though after reading getpelican/cookiecutter-pelican-plugin#4 I wonder if it's better to wait.

**Comment by justinmayer** [comment]
This PR now contains only the `pyproject.toml` changes needed to install the plugin, which of course is all this PR should have contained from the beginning. My apologies for the previous noise and clutter.

@davidag: Now that this PR has been appropriately streamlined, does everything look okay to you?

**Comment by justinmayer** [comment]
Since this plugin can't be installed without merging these changes, I'm going to go ahead and merge this PR.

**Comment by davidag** [comment]
> Since this plugin can't be installed without merging these changes, I'm going to go ahead and merge this PR.

@justinmayer sorry for the delay merging this, I was trying to verify the whole new plugin development setup but I came across getpelican/pelican#2653 and had to stop, but I planned to continue in this coming week. I'll review it anyway even if only to understand the whole setup.

I wasn't aware that without this changes the plugin couldn't be installed. In getpelican/cookiecutter-pelican-plugin#3 I only reported not being able to run the tests. Because the new plugin structure support was not implemented yet, I couldn't have tested it. Maybe you're already trying @avaris changes in getpelican/pelican#2644?

**Comment by justinmayer** [comment]
Sorry for not being clearer. I don't remember exactly since it's been a while, but it's possible that I was referring to the new plugin structure. I was indeed testing with the changes in PR 2644, yes.

**Comment by davidag** [comment]
I've been doing some testing and tests fail to run even with this changes merged. I'm trying some options using poetry but it's not working yet...

**Pull request merged**